content/v2.0/application/booting.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Booting
+order: 40
+---
+
+Hanami applications support a **prepared** state and a **booted** state.
+
+Whether your app is prepared or booted determines whether its components are **lazily loaded** on demand (prepared stated), or **eagerly loaded** up front (booted state).
+
+This difference may not sound like much, but the ability of a prepared app to load _just enough_ of the app to undertake a particular task can be a huge performance advantage when it comes to testing, worker processes and other use cases where your app doesn't need every component to perform the job at hand.
+
+Conversely, a booted app is fully initialized, with every component ready to do its work. This is ideal whenever you want to incur initialization costs at boot time, such as when serving web requests.
+
+## Hanami.prepare
+
+When you call `Hanami.prepare` (or use `require "hanami/prepare"`), Hanami will prepare your app for use.
+
+This process includes defining various Ruby constants required for your app to function, making the Ruby source files in `app/` autoloadable, and making your components (the classes you've added to the `app/` folder) lazily loadable.
+
+This approach keeps load time to a minimum. As such, it’s the default mode in the Hanami console and when running tests.
+
+<p class="notice">
+  A prepared application is no less able to do its job than a booted one, it will just lazily load only the components it needs to perform a particular task on demand.
+</p>
+
+## Hanami.boot
+
+When you call `Hanami.boot` (or use `require "hanami/boot"`) Hanami will do everything it does when `Hanami.prepare` is called, but then go further, starting each of your app's providers, while also eagerly loading all of your app's components up front.
+
+This is useful when you want to incur all initialization costs at boot time, such as when preparing an app to serve web requests.
+
+Booting is the approach taken in Hanami's standard Puma setup. Thus, in Hanami's `config.ru` file you will see:
+
+```ruby
+# frozen_string_literal: true
+
+require "hanami/boot"
+
+run Hanami.app
+```
+
+## Stepping through the boot process
```

Comment:
I really love this section! Does a lot to demystify the boot process ❤️

Reply:
I learnt a few things :)